File: tools/seed_firebase_test_data.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/widgets.dart';
import 'package:quit_smoking_club/firebase_options.dart';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);

  final docId = 'Medical|all|Firebase Test Item';
  final data = {
    'category': 'Medical',
    'language': 'all',
    'title': 'Firebase Test Item',
    'content': 'This item was created to verify Firebase connectivity.',
    'link': 'https://firebase.google.com',
    'unique_id': docId,
  };

  await FirebaseFirestore.instance
      .collection('content_items')
      .doc(docId)
      .set(data);

  final snapshot = await FirebaseFirestore.instance
      .collection('content_items')
      .doc(docId)
      .get();

  if (!snapshot.exists) {
    throw Exception('Seeded document was not found after write.');
  }

  print('✅ Firebase seed completed.');
  print(snapshot.data());
}

File: tools/seed_firebase_test_data.py
import os
import sys

try:
    import firebase_admin
    from firebase_admin import credentials, firestore
except Exception as e:
    print('Missing firebase_admin package:', e)
    sys.exit(1)

SERVICE_ACCOUNT_PATH = os.path.join(os.getcwd(), 'firebase-service-account.json')

if not os.path.exists(SERVICE_ACCOUNT_PATH):
    print('Missing service account file:', SERVICE_ACCOUNT_PATH)
    sys.exit(1)

cred = credentials.Certificate(SERVICE_ACCOUNT_PATH)
firebase_admin.initialize_app(cred)

db = firestore.client()
doc_id = 'Medical|all|Firebase Test Item'
data = {
    'category': 'Medical',
    'language': 'all',
    'title': 'Firebase Test Item',
    'content': 'This item was created to verify Firebase connectivity.',
    'link': 'https://firebase.google.com',
    'unique_id': docId,
}

db.collection('content_items').document(doc_id).set(data)
print('✅ Firebase seed completed.')
